--- ../src-base/minecraft/net/minecraft/block/BlockShulkerBox.java
+++ ../src-work/minecraft/net/minecraft/block/BlockShulkerBox.java
@@ -32,6 +32,8 @@
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
 
+import carpet.CarpetSettings;
+
 public class BlockShulkerBox extends BlockContainer
 {
     public static final PropertyEnum<EnumFacing> field_190957_a = PropertyDirection.func_177714_a("facing");
@@ -99,6 +101,11 @@
                     flag = true;
                 }
 
+                //creativeOpenContainerNoCheck
+                if (CarpetSettings.creativeOpenContainerNoCheck && p_180639_4_.func_184812_l_()) {
+                    flag = true;
+                }
+
                 if (flag)
                 {
                     p_180639_4_.func_71029_a(StatList.field_191272_ae);
@@ -176,7 +183,7 @@
                 NBTTagCompound nbttagcompound = new NBTTagCompound();
                 NBTTagCompound nbttagcompound1 = new NBTTagCompound();
                 nbttagcompound.func_74782_a("BlockEntityTag", ((TileEntityShulkerBox)tileentity).func_190580_f(nbttagcompound1));
-                itemstack.func_77982_d(nbttagcompound);
+                if(!CarpetSettings.stackableEmptyShulkerBoxes || nbttagcompound1.func_186856_d() > 0) itemstack.func_77982_d(nbttagcompound);
 
                 if (tileentityshulkerbox.func_145818_k_())
                 {
